ivy/array/general.py

```diff
@@ -1038,24 +1038,38 @@ def inplace_update(
         return ivy.inplace_update(self, val, ensure_in_backend=ensure_in_backend)
 
     def inplace_increment(
-        self: ivy.Array, val: Union[ivy.Array, ivy.NativeArray]
+        self: Union[ivy.Array, ivy.NativeArray], val: Union[ivy.Array, ivy.NativeArray]
```

Comment:
The original type hints for `self` (`ivy.Array`) is correct as this is an instance method for ivy.Array, so the type must be ivy.Array only